derivers.lua

```diff
@@ -119,18 +169,62 @@ local eq = {
 		assert(compiled, message)
 		local eq_fn = compiled(eq_memoizer)
 		t.__eq = eq_fn
+
+		t.derived_eq = true
+	end,
+	foreign = function()
+		error("can't derive eq for a foreign type")
```

Comment:
these should say "can't derive eq()" for clarity. This applies to all "can't derive blah" statements in this file.